apache/airflow
# [AIRFLOW-85] Convert Airflow Webserver UI to FAB for RBAC support

Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-85


### Description
New UI with RBAC support.

### Tests
- See tests/www_rbac/

### Commits
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

To test this PR:
1. Check out this PR locally
2. Rerun `python setup.py install` or `python setup.py develop` so Flask-AppBuilder at [**master**](https://github.com/dpgaspar/Flask-AppBuilder) can be installed.
3. In your `airflow.cfg` file, under `[webserver]`, set `rbac = True`.  
5. Run `airflow initdb` to generate new tables for RBAC support.
6. Run `airflow create_admin` to create an admin user.
7. Run `airflow webserver` as usual.

Note that the default auth type is DB.  You can change AUTH_TYPE in the newly generated `webserver_config.py` file in AIRFLOW_HOME. See [this link](http://flask-appbuilder.readthedocs.io/en/latest/security.html#authentication-methods) for instructions.

Given the size of this PR, it will likely require a few iterations to get the rough edges sorted out. Love any help I can get with code-review and testing.
